src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -323,12 +323,12 @@ async function signAndPush(req, res) {
     const transaction = atomicSwapService.assembleTransaction(partialTx, sigs, req.wallet.storage);
 
     const sendTransaction = new SendTransaction({ transaction, storage: req.wallet.storage });
-    const response = await sendTransaction.runFromMining();
+    const response = await runSendTransaction(sendTransaction, unlock);
     res.send({ success: true, ...mapTxReturn(response) });
   } catch (err) {
     res.send({ success: false, error: err.message });
   } finally {
-    lock.get(req.walletId).unlock(lockTypes.SEND_TX);
+    unlock();
```

Comment:
The `unlock` method is always called in case there are no errors inside the `runSendTransaction` method. Should we move this call inside the `catch` block?

If yes, we should apply to all controllers.